client/src/hoc/withResizeHandler.js

```diff
@@ -0,0 +1,31 @@
+// This component uses a resize listener event to update itself and whatever component it is wrapped around.
+import { Component } from 'react';
+
+// eslint-disable-next-line
+const withResizeHandler = (WrappedComponent) => {
+  return class extends Component {
+    // _isMounted (and handling it in the mounted methods) is to prevent setting state on an unmounted component
+    _isMounted = false;
```

Comment:
Could you elaborate on the decision to make this hook?

Reply:
If you mean the _isMounted hook, I kept getting an error that it was trying to set state on an unmounted component and it might indicate a memory leak. I was confused since I removed the event listener on unmount, but I found a solution online that said you can handle this hook and it would work. I still get the error, so I can remove it. I don't understand why it is throwing the error though.